src/Xamarin.Android.Build.Tasks/Tasks/GenerateMainAndroidManifest.cs

```diff
@@ -68,21 +58,18 @@ public override bool RunTask ()
 		var userAssembliesPerArch = MonoAndroidHelper.GetPerArchAssemblies (ResolvedUserAssemblies, SupportedAbis, validate: true);
 
 		androidRuntime = MonoAndroidHelper.ParseAndroidRuntime (AndroidRuntime);
-		codeGenerationTarget = MonoAndroidHelper.ParseCodeGenerationTarget (CodeGenerationTarget);
 
 		// Generate the merged manifest
 		var additionalProviders = MergeManifest (templateCodeGenState, GenerateJavaStubs.MaybeGetArchAssemblies (userAssembliesPerArch, templateCodeGenState.TargetArch));
-		GenerateAdditionalProviderSources (templateCodeGenState, additionalProviders);
 
+		AdditionalProviderSources = additionalProviders.Select (p => new TaskItem (p)).ToArray ();
```

Comment:
Could we change the `AdditionalProviderSources` property to `string[]` and then this could be:
```suggestion
		AdditionalProviderSources = additionalProviders.ToArray ();
```
If it's just used by MSBuild via `[Output]`, it doesn't have to be the `ITaskItem` type. You'd really only need that if you also set item metadata.

Reply:
TIL: I didn't realize it supported arrays other than `ITaskItem`, nice!